### PR TITLE
Add functions for checking plot labels

### PR DESCRIPTION
**Pull request by rossellhayes**
Adds function `get_labels()` for listing (a subset of) the labels of a plot, and `uses_labels()` for checking if labels match their expected values.

``` r
library(ggcheck)
library(ggplot2)

p <- ggplot(mtcars, aes(x = wt, y = mpg, color = cyl)) +
  geom_point() +
  labs(x = "weight", y = "MPG", color = NULL)

get_labels(p)
#> $x
#> [1] "weight"
#> 
#> $y
#> [1] "MPG"
#> 
#> $colour
#> NULL

get_labels(p, c("x", "y"))
#> $x
#> [1] "weight"
#> 
#> $y
#> [1] "MPG"

uses_labels(p, x = "weight")
#> [1] TRUE

uses_labels(p, color = NULL)
#> [1] TRUE

# If multiple arguments are given, returns TRUE only if all labels match
uses_labels(p, x = "weight", color = NULL)
#> [1] TRUE
```

<sup>Created on 2021-11-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### CODE REVIEW

**Review by deirdrefrancks — COMMENTED** [review]
Looks good to me!

File: R/labels.R [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    stop(
+      "You must pass an argument to `...` in `uses_labels()`.",
+      call. = FALSE
+    )
+  }
+
+  if (
+    !all(vapply(args, length, integer(1)) <= 1) ||
+      !all(vapply(args[lengths(args) > 0], is.character, logical(1))) ||
+      !all(vapply(args[lengths(args) == 0], is.null, logical(1)))
```
Comment (gadenbuie) [inline R/labels.R:86]: Let's use `rlang` or refactor this into a helper function or intermediate variables

Comment (rossellhayes) [reply R/labels.R]: Addressed in 7a5677c

**Review by gadenbuie — APPROVED** [review]
LGTM!

**Pull request merged**